src/hivpy/experiment.py

```diff
@@ -1,10 +1,58 @@
 import os
-from datetime import date, timedelta
+from datetime import date, datetime, timedelta
 
-from .config import ExperimentConfig, OutputConfig, SimulationConfig
+import numpy as np
+import pandas as pd
+
+from .config import ExperimentConfig, LoggingConfig, SimulationConfig
+from .exceptions import SimulationException
 from .simulation import run_simulation
 
 
+class OutputHandler:
+    """
+    Handles output for the experiment.
+    Could be used to just keep track of files and handle writing/reading
+    Or could also be used to calculate summary statistics
+    Although if that is going to be elaborate, a separate statistics module
+    could be useful.
+    """
+    files: dict
+    output_dir: str
+    simulation_data: pd.DataFrame
+    summary_stats: np.ndarray
+
+    def __init__(self, output_params):
+        self.output_dir = os.path.normpath(output_params['OUTPUT_DIR'])
+        self.files = {"Raw": "raw_data.out", "Stats": "summary_statistics.out"}
+
+    def set_file_name(self, fileId, filename):
+        filepath = os.path.join(self.output_dir, filename)
+        self.files[fileId] = filepath
+
+    def _commit_data(self, fileId, data, write_mode):
+        if(not (fileId in self.files)):
+            raise SimulationException(f"File matching {fileId} does not exist.")
+        if(isinstance(data, pd.DataFrame)):
+            data.to_csv(self.files[fileId], mode=write_mode)
+        elif(isinstance(data, np.ndarray)):
+            outfile = open(self.files[fileId], write_mode)
+            np.savetxt(outfile, data)
+            outfile.close()
```

Comment:
Can use a context manager to handle closing of the file at the end or if any errors occur:
```suggestion
            with open(self.files[fileId], write_mode) as outfile:
                np.savetxt(outfile, data)
```
Although here there's not much to protect against.

Reply:
Do we want this to also be in CSV? If so, we should pass `delimiter=","` to `np.savetxt` (I think it uses spaces by default)

Reply:
Agreed; I should really have put in a try statement in case the file write went wrong somehow, but a context manager is neater. (I find them a bit confusing because I don't know what classes have __enter__ and __exit__ functionalities to allow them to be used with it.) Also yes it is a good idea to keep the files consistent so I'll make it comma delimited!